nemo/collections/nlp/models/language_modeling/megatron_retrieval_model.py

```diff
@@ -86,7 +89,43 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
         else:
             raise ValueError('precision must be in [32, 16, "bf16"]')
         self.model.model_type = ModelType.encoder_and_decoder
-        # self.grad_clip_pl_default = True
+
+        if hasattr(self.cfg, "shape_file"):
+            # self.grad_clip_pl_default = True
+            set_base_shapes(self, self.cfg.shape_file, rescale_params=False)
+            # add shape file
+            self.register_artifact("model.shape_file", self.cfg.shape_file),
+
+            for name, tensor in self.named_parameters():
```

Comment:
Can you add a comment explaining this initialization scheme?

Reply:
added